modules/OFConnectionManager2/module/src/bundle.c

```diff
@@ -628,14 +634,18 @@ bundle_task(void *cookie)
                 return IND_SOC_TASK_CONTINUE;
             }
         }
+   
         /* clean up subbundle */
-        aim_free(subbundle->objs);
-        subbundle->objs = NULL;
-        /* invoke subbundle finish before moving onto next subbundle */
-        subbundle_info.subbundle_idx = state->cur_subbundle;
-        subbundle_info.total_msg_count = state->subbundles[state->cur_subbundle].count;
-        subbundle_info.is_aborted = state->is_aborted;
-        invoke_subbundle_finish(state->cxn_id, &subbundle_info);
+        if (subbundle->objs) {
+            aim_free(subbundle->objs);
+            subbundle->objs = NULL;
+            /* invoke subbundle finish before moving onto next subbundle */
+            subbundle_info.subbundle_idx = state->cur_subbundle;
+            subbundle_info.total_msg_count = state->subbundles[state->cur_subbundle].count;
+            subbundle_info.is_aborted = state->is_aborted;
+            invoke_subbundle_finish(state->cxn_id, &subbundle_info);
+        }
```

Comment:
if i understood the core correctly, it looks like could invoke "start" without a corresponding "finish" when no objects are assigned to a subbundle; let's aim for each "start" to have a corresponding "finish".

Reply:
I will skip the start() when there is no object in the subbundle.